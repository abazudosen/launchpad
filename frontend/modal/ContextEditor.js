/* @flow */
/**
 * Edits the secrets for a pad
 */

import React, { Component } from 'react';
import type { Context } from './types';
import KeyValueEditor from './KeyValueEditor';
import './ContextEditor.less';

export default class ContextEditor extends Component {
  props: {|
    context: Array<Context>,
    onChange: (contexts: Array<Context>) => any,
  |};

  render() {
    return (
      <div className="ContextEditor">
        <KeyValueEditor
          value={this.props.context}
          onChange={this.props.onChange}
          hilightEmpty
        />
      </div>
    );
  }
}
