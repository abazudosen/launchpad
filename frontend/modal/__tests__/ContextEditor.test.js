/* @flow */

import React from 'react';
import ContextEditor from '../ContextEditor';
import renderer from 'react-test-renderer';

test('renders empty', () => {
  const component = renderer.create(
    <ContextEditor context={[]} onChange={() => {}} />,
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('renders hidden', () => {
  const component = renderer.create(
    <ContextEditor
      context={[
        {
          key: 'test',
        },
        {
          key: 'test2',
        },
      ]}
      onChange={() => {}}
    />,
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('renders normal', () => {
  const component = renderer.create(
    <ContextEditor
      context={[
        {
          key: 'test',
          value: 'value',
        },
        {
          key: 'test2',
          value: 'value2',
        },
      ]}
      onChange={() => {}}
    />,
  );
  expect(component.toJSON()).toMatchSnapshot();
});
