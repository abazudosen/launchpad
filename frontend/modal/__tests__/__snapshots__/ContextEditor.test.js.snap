// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders empty 1`] = `
<div
  className="ContextEditor"
>
  <div
    className="KeyValueEditor"
  >
    <div
      className="KeyValueEditor-Line"
    >
      <div
        className="KeyValueEditor-Label"
      >
        Key
      </div>
      <div
        className="KeyValueEditor-Label"
      >
        Value
      </div>
    </div>
    <button
      className="KeyValueEditor-AddButton btn primary"
      onClick={[Function]}
    >
      <span>
        Add
      </span>
      <span
        className="icon-plus"
      />
    </button>
  </div>
</div>
`;

exports[`renders hidden 1`] = `
<div
  className="ContextEditor"
>
  <div
    className="KeyValueEditor"
  >
    <div
      className="KeyValueEditor-Line"
    >
      <div
        className="KeyValueEditor-Label"
      >
        Key
      </div>
      <div
        className="KeyValueEditor-Label"
      >
        Value
      </div>
    </div>
    <div
      className="KeyValueEditor-Line"
    >
      <input
        className="KeyValueEditor-Input"
        disabled={false}
        onChange={[Function]}
        value="test"
      />
      <input
        className="KeyValueEditor-Input"
        disabled={true}
        value="*****"
      />
      <button
        className="KeyValueEditor-DeleteButton"
        onClick={[Function]}
      >
        <span
          className="icon-cross"
        />
      </button>
    </div>
    <div
      className="KeyValueEditor-Line"
    >
      <input
        className="KeyValueEditor-Input"
        disabled={false}
        onChange={[Function]}
        value="test2"
      />
      <input
        className="KeyValueEditor-Input"
        disabled={true}
        value="*****"
      />
      <button
        className="KeyValueEditor-DeleteButton"
        onClick={[Function]}
      >
        <span
          className="icon-cross"
        />
      </button>
    </div>
    <button
      className="KeyValueEditor-AddButton btn primary"
      onClick={[Function]}
    >
      <span>
        Add
      </span>
      <span
        className="icon-plus"
      />
    </button>
  </div>
</div>
`;

exports[`renders normal 1`] = `
<div
  className="ContextEditor"
>
  <div
    className="KeyValueEditor"
  >
    <div
      className="KeyValueEditor-Line"
    >
      <div
        className="KeyValueEditor-Label"
      >
        Key
      </div>
      <div
        className="KeyValueEditor-Label"
      >
        Value
      </div>
    </div>
    <div
      className="KeyValueEditor-Line"
    >
      <input
        className="KeyValueEditor-Input"
        disabled={false}
        onChange={[Function]}
        value="test"
      />
      <input
        className="KeyValueEditor-Input"
        onChange={[Function]}
        value="value"
      />
      <button
        className="KeyValueEditor-DeleteButton"
        onClick={[Function]}
      >
        <span
          className="icon-cross"
        />
      </button>
    </div>
    <div
      className="KeyValueEditor-Line"
    >
      <input
        className="KeyValueEditor-Input"
        disabled={false}
        onChange={[Function]}
        value="test2"
      />
      <input
        className="KeyValueEditor-Input"
        onChange={[Function]}
        value="value2"
      />
      <button
        className="KeyValueEditor-DeleteButton"
        onClick={[Function]}
      >
        <span
          className="icon-cross"
        />
      </button>
    </div>
    <button
      className="KeyValueEditor-AddButton btn primary"
      onClick={[Function]}
    >
      <span>
        Add
      </span>
      <span
        className="icon-plus"
      />
    </button>
  </div>
</div>
`;
